### PR TITLE
utils_test/__init__: Fix autotest issues inside SLES guests

### PR DESCRIPTION
**Pull request by ngu-niny**
1. Fix failure to get 'autotest_dirname' issue: Add lines in function 'extract(vm, remote_path, dest_dir)' to decide whether the 'extracting autotest.tar.bz2 output' and 'autotest_dirname' is null, if so, try to continue to get it in the remained 'while/for' loops; otherwise, break and go on with following test
2. Fix the 'cat' results in host failure issue: Add a 'while' loop in function 'get_results(base_results_dir)' to make sure file 'status.json' already created before compressing the result

ID: 1360676, 1361468

Signed-off-by: Nini Gu ngu@redhat.com

### CODE REVIEW

File: virttest/utils_test/__init__.py [hunk 1]
```diff
@@ -790,6 +791,7 @@ def get_results(base_results_dir):
         compress_cmd += "tar cjvf %s ./*" % results_tarball
         compress_cmd += " --exclude=*core*"
         compress_cmd += " --exclude=*crash*"
+        time.sleep(5)
```
Comment (PandaWei) [inline virttest/utils_test/__init__.py]: Frankly, I hate the abuse of sleep().

**Comment by ngu-niny** [comment]
@PandaWei Do you have any suggestion to resolve this type of issue without using sleep()? 
Thanks!

**Comment by PandaWei** [comment]
what's mean that ''Fix the 'cat' autotest results failure issue"?

**Comment by PandaWei** [comment]
If you worry about that compress operation happens before test is still runing and writing the log, 
You should ensure that Any tests in guest had already finished before calling get_result().

**Comment by PandaWei** [comment]
I had a discussion with @balamuruhans for the similar problem. #664 
calling sleep() in API function is not a good choice.

**Comment by ngu-niny** [comment]
@PandaWei Thanks a lot for above info. I have updated the patch, would you help to review it again?

File: virttest/utils_test/__init__.py [hunk 2]
```diff
@@ -790,7 +795,12 @@ def get_results(base_results_dir):
         compress_cmd += "tar cjvf %s ./*" % results_tarball
         compress_cmd += " --exclude=*core*"
         compress_cmd += " --exclude=*crash*"
-        session.cmd(compress_cmd, timeout=600)
+        while True:
+            status_json = os.path.join(results_dir, "status.json")
+            if session.cmd_status("test -f %s" % status_json) != 0:
+                continue
```
Comment (CongLi) [inline virttest/utils_test/__init__.py]: It has been done in get_results_summary().

File: virttest/utils_test/__init__.py [hunk 3]
```diff
@@ -755,11 +755,16 @@ def extract(vm, remote_path, dest_dir):
         extract_cmd = "tar %s %s -C %s" % (options,
                                            basename,
                                            os.path.dirname(dest_dir))
-        output = session.cmd(extract_cmd, timeout=120)
+        output = ""
+        while True:
+            output = session.cmd(extract_cmd, timeout=120)
+            if output != "":
+                break
```
Comment (PandaWei) [inline virttest/utils_test/__init__.py]: IMHO, There's no obvious improvement than sleep().
Even this method is worse, the loop maybe endless.
session.cmd() return means the extract command had been executed and finished. If the output is not your expectation, You should investigate why and find solution.  

If it's impossible to find the true reason and there's no better way to fix it, but by sleep().  I'm OK , but please add explanation for it.

I think you should investigate why

**Comment by ngu-niny** [comment]
@PandaWei For the first 'extracting autotest.tar.bz2 output' failure issue, sleep() even 300 seconds could not help to resolve it. With my current resolution, I have added a sentence 'print "####output%d: %s" % (i, output)' to do some debug; from following test log output, it can be seen the 1st time running cmd 'output = session.cmd(extract_cmd, timeout=120)' gets null output, WHILE the session log showed the 1st time 'extract' cmd was already correctly executed. Not sure about the real reason now. Hence, shall we add a for loop instead of the possible endless while one?

02:07:10 DEBUG| Sending command: tar xvjmf autotest.tar.bz2 -C /usr/local
02:07:10 DEBUG| ####output1:
02:07:10 DEBUG| Sending command: tar xvjmf autotest.tar.bz2 -C /usr/local
02:07:11 INFO | Update MAC(34:40:b5:a0:ca:10)<->(10.73.196.127)IP pair into address_cache
02:07:11 DEBUG| ####output2: client/__init__.py
02:07:11 DEBUG| client/autotest_local.py
02:07:11 DEBUG| client/base_sysinfo.py
02:07:11 DEBUG| client/base_utils.py
02:07:11 DEBUG| client/base_utils_unittest.py

For the second 'cat' results in host failure issue, usually adding sleep(10) could help to resolve it.

**Comment by PandaWei** [comment]
Yes, for loop is better than endless while.

File: virttest/utils_test/__init__.py (hunk 3 above)
Comment (xutian) [inline virttest/utils_test/__init__.py]: @ngu-niny  maybe '-v' or '--verbose' is helpful for you.

File: virttest/utils_test/__init__.py [hunk 4]
```diff
@@ -790,7 +795,12 @@ def get_results(base_results_dir):
         compress_cmd += "tar cjvf %s ./*" % results_tarball
         compress_cmd += " --exclude=*core*"
         compress_cmd += " --exclude=*crash*"
-        session.cmd(compress_cmd, timeout=600)
+        while True:
```
Comment (xutian) [inline virttest/utils_test/__init__.py]: Agree with @PandaWei 's comments, you should find out root cause of output of extra_cmd is not expect, not repeat, repeat and repeat...

**Comment by xutian** [comment]
so long not update, let me close it. @ngu-niny please find out rootcause then send a new pull-request.

**Comment by ngu-niny** [comment]
New PR is https://github.com/avocado-framework/avocado-vt/pull/949

**Pull request closed without merging**